nsxt/resource_nsxt_ip_block.go

```diff
@@ -0,0 +1,337 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/manager"
+	"log"
+	"net/http"
+)
+
+// formatLogicalRouterRollbackError defines the verbose error when
+// rollback fails on a logical router create operation.
+const formatIpBlockRollbackError = `
+WARNING:
+There was an error during the creation of IP block router %s:
+%s
+Additionally, there was an error deleting the IP block during rollback:
+%s
+The IP block may still exist in the NSX. If it does, please manually delete it
+and try again.
+`
+
+func resourceNsxtIpBlock() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtIpBlockCreate,
+		Read:   resourceNsxtIpBlockRead,
+		Update: resourceNsxtIpBlockUpdate,
+		Delete: resourceNsxtIpBlockDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"cidr": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Represents network address and the prefix length which will be associated with a layer-2 broadcast domain",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"subnet": &schema.Schema{
+				Type:        schema.TypeList,
+				Description: "List of subnets",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
```

Comment:
I think we need to understand where these ids are used.. if other resources could be dependent on these, maybe it would be preferable to expose these as separate objects?